k2/python/k2/fsa.py

```diff
@@ -1029,44 +1029,23 @@ def from_openfst(cls, s: str, acceptor: bool = True) -> 'Fsa':
         arcs, aux_labels = _k2.fsa_from_str(s, acceptor, True)
         return Fsa(arcs, aux_labels=aux_labels)
 
-    def set_scores_stochastic_(self) -> None:
-        '''Set `scores` to random numbers.
+    def set_scores_stochastic_(self, scores) -> None:
+        '''Normalize the given `scores` and assign it to self.
```

Comment:
I'd write this as:
   Normalize the given `scores` and assign the result to self.scores.
 Args:
     scores:   Tensor of scores of dtype torch.float32, and shape equal to self.scores.shape
            (one axis).  Will be normalized so the sum, after exponentiating, of the scores 
            leaving each state that has at least one arc leaving it is 1.

Reply:
thanks, changed.